types/draft-js/index.d.ts

```diff
@@ -852,9 +852,21 @@ declare namespace Draft {
                 hasText(): boolean;
             }
 
+            interface SelectionStateProperties {
+                anchorKey: string
+                anchorOffset: number
+                focusKey: string
+                focusOffset: number
+                isBackward: boolean
+                hasFocus: boolean
+            }
+
             class SelectionState extends Record {
                 static createEmpty(key: string): SelectionState;
 
+                merge(...iterables: Immutable.Iterable<keyof SelectionStateProperties, SelectionStateProperties[keyof SelectionStateProperties]>[]): SelectionState
```

Comment:
The type of `...iterables` here is not great I will admit, but it was the best I could find that matches the signature of the superclass.

Based on the docs (and my own usage) the version that takes a JS object is more commonly used anyway.